### PR TITLE
Removed deprecated methods and classes

### PR DESCRIPTION
**Pull request by ahcorde**
Most of the deprecated happened since `iron`.

Since Jazzy:

 - rate
 - typesupport_helpers
 - waitable

### CODE REVIEW

**Review by alsora — APPROVED** [review]

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21433/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15723/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/932/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22154/)

File: rclcpp/include/rclcpp/client.hpp [hunk 1]
```diff
@@ -423,6 +415,7 @@ class Client : public ClientBase
 
   RCLCPP_SMART_PTR_DEFINITIONS(Client)
 
+
```
Comment (fujitatomoya) [inline rclcpp/include/rclcpp/client.hpp:418]: nit: unrelated change?

**Review by fujitatomoya — APPROVED** [review]
lgtm with green CI

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21489/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15773/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/989/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22204/)
